sdks/go/pkg/beam/core/metrics/metrics_test.go

```diff
@@ -25,6 +25,44 @@ import (
 // bID is a bundleId to use in the tests, if nothing more specific is needed.
 const bID = "bID"
 
+// TestRobustness validates metrics not panicking if the context doesn't
+// have the bundle or transform ID.
+func TestRobustness(t *testing.T) {
+	ns, n := "Test", "myCount"
+	m := NewCounter(ns, n)
+	m.Inc(context.Background(), 3)
+	ptCtx := SetPTransformID(context.Background(), "MY_TRANSFORM")
+	m.Inc(ptCtx, 3)
+	bCtx := SetBundleID(context.Background(), "MY_BUNDLE")
```

Comment:
Should probably use bID here to be consistent with the other tests.

Reply:
Done. Also removed the variables for namespace and name (ns, n) because they weren't re-used in this small test.